### ISSUE TITLE
Blue ocean not loading

### ISSUE THREAD

**Issue by T1mey** [SAP/devops-docker-cx-server#103]
![image](https://user-images.githubusercontent.com/10464878/114359587-681a5580-9b74-11eb-8381-1b598301680c.png)

**Comment by fwilhe** [comment]
Hi @T1mey 

thanks for reporting.

Which version of cx-server are you using? What happened before (version update of cx-server?) before this error came up?

**Comment by T1mey** [comment]
We're using the newst build v58.
But this issue is there for some time now. Not sure when it has been started to show that.

**Comment by fwilhe** [comment]
Ok, but the blue ocean ui itself does work, right?

We don't install this jira plugin via our [plugins.txt](https://github.com/SAP/devops-docker-cx-server/blob/master/jenkins-master/plugins.txt), not sure how this dependency is defined.

**Comment by T1mey** [comment]
Strange. We have installed only 3 plugins

* Fortify
* Role based strategy and
* SAML
None of them has this dependency.

**Comment by fwilhe** [comment]
I guess that cx-server does not automatically update plugins in that case as documented [here](https://github.com/SAP/devops-docker-cx-server/blob/master/docs/operations/cx-server-operations-guide.md#plugins). I just checked with my old docker volume, and I could reproduce your issue, with a clean volume I did not encounter that issue.

My understanding is that `cx-server update` will not touch your plugins as soon as one custom plugin is installed, so that leaves the task of updating plugins to you as the administrator.

An alternative approach might be to remove the plugins in the docker volume (be sure to take a [backup](https://github.com/SAP/devops-docker-cx-server/blob/master/docs/operations/cx-server-operations-guide.md#backup) first), update cx server to get the latest plugins, and reinstall your custom plugins after that.

Hope this helps

Florian

**Comment by T1mey** [comment]
So let me summarize my understanding.

the documentation says that only the "standard" plugins are updated by a cx-server update. If I use custom one's I'm responsible for updating them (ond their dependencies). 
Fine with that.

On the other side all "standard" plugins should be updated by cx-server update script. 

As I only have 3 plugins installed. In my update list a lot of entries are not related to my plugins. Why aren't they updated?

![image](https://user-images.githubusercontent.com/10464878/114503693-fe5e8200-9c2d-11eb-89b8-07ab69f72c47.png)

**Comment by fwilhe** [comment]
Did you ever update a plugin manually? Because if so, this plugin gets thrown out of the update-cycle as documented [in the Jenkins docs](https://github.com/jenkinsci/docker/blob/master/README.md#upgrading-plugins).

Otherwise, I'm not sure if I see how many updates are available in your system in that screenshot.

v59 was released yesterday, can you check if that issue persists?

**Comment by T1mey** [comment]
I never updated a Plugin manually.
I installed the Role based auth strategy Plugin before it was added to the Standard.

**Comment by T1mey** [comment]
updated to version v59. No update of any plugin was done.

**Comment by T1mey** [comment]
@fwilhe any idea ?

**Comment by fwilhe** [comment]
Not more that what I already said.

To summarize:

Plugins should be automatically updated if you don't "opt out" by manually updating. Installing additional plugins is fine. What you tell me sounds like this automatic life-cycle mgmt is broken, I don't know how this did happen.

The way forward would be to remove the plugins from jenkins home (then Jenkins would re-populate that directory with the plugins shipped via the image) and then to add your custom plugins.

Sorry that I can't provide a more straightforward solution, but this is essentially nothing special about CX-Server, it is just plain Jenkins-docker.

Also, in any case make sure to have an up-to-date backup of jenkins home (for example via `cx-server backup`) just to be safe.

Regards

Florian

**Comment by T1mey** [comment]
So we finally deleted the complete plugins folder and Jenkins recreated it. 
So blue ocean is working again, because that plugin which causes the troble isn't installed anymore. 
But as I said we haven't installed it manually or with any other plugin as dependency.

As we have our own Dockerfile we added our plugins to a seperate plugins.txt file and do the installation as your are doing that in yours in addition.

Doing so we realized why there is that long list of updates.  install-plugins.sh is not updating to the latest version of the plugin. This depends on the used jenkins version. So it's updating to the version which available for the Jenkins version.
